yoti-sdk-impl/src/test/java/com/yoti/api/client/spi/remote/HumanProfileAdapterTest.java

```diff
@@ -59,4 +81,80 @@ public void isAgeVerified_shouldPreferAgeOverWhenBothAttributesArePresent() {
         assertTrue(result);
     }
 
+    @Test
+    public void getAgeVerifications_shouldBeNullSafe() {
+        when(profileMock.findAttributesStartingWith(HumanProfileAttributes.AGE_UNDER, String.class)).thenReturn(EMPTY_LIST);
+        when(profileMock.findAttributesStartingWith(HumanProfileAttributes.AGE_OVER, String.class)).thenReturn(EMPTY_LIST);
+
+        List<AgeVerification> result = testObj.getAgeVerifications();
+
+        assertThat(result, hasSize(0));
+    }
+
+    @Test
+    public void getAgeVerifications_shouldOnlySearchWrappedProfileOnce() {
+        when(profileMock.findAttributesStartingWith(HumanProfileAttributes.AGE_UNDER, String.class)).thenReturn(ageUnderAttributes);
+        when(profileMock.findAttributesStartingWith(HumanProfileAttributes.AGE_OVER, String.class)).thenReturn(ageOverAttributes);
+
+        testObj.getAgeVerifications();
+        List<AgeVerification> result = testObj.getAgeVerifications();
+
+        assertThat(result, hasSize(4));
+        assertThat(new HashSet<>(result), hasSize(4));
+        assertThat(result.get(0).getAttribute(), isOneOf(ageUnder18Attribute, ageUnder21Attribute, ageOver18Attribute, ageOver21Attribute));
+        assertThat(result.get(1).getAttribute(), isOneOf(ageUnder18Attribute, ageUnder21Attribute, ageOver18Attribute, ageOver21Attribute));
+        assertThat(result.get(2).getAttribute(), isOneOf(ageUnder18Attribute, ageUnder21Attribute, ageOver18Attribute, ageOver21Attribute));
+        assertThat(result.get(3).getAttribute(), isOneOf(ageUnder18Attribute, ageUnder21Attribute, ageOver18Attribute, ageOver21Attribute));
+        verify(profileMock, times(1)).findAttributesStartingWith(HumanProfileAttributes.AGE_UNDER, String.class);
+        verify(profileMock, times(1)).findAttributesStartingWith(HumanProfileAttributes.AGE_OVER, String.class);
+    }
+
+    @Test
+    public void findAgeOverVerification_returnsNullForNoMatch() {
+        when(profileMock.findAttributesStartingWith(HumanProfileAttributes.AGE_UNDER, String.class)).thenReturn(EMPTY_LIST);
+        when(profileMock.findAttributesStartingWith(HumanProfileAttributes.AGE_OVER, String.class)).thenReturn(EMPTY_LIST);
+
+        testObj.findAgeOverVerification(18);
+        AgeVerification result = testObj.findAgeOverVerification(18);
+
+        assertNull(result);
+    }
+
+    @Test
+    public void findAgeOverVerification_shouldOnlySearchWrappedProfileOnce() {
+        when(profileMock.findAttributesStartingWith(HumanProfileAttributes.AGE_UNDER, String.class)).thenReturn(ageUnderAttributes);
+        when(profileMock.findAttributesStartingWith(HumanProfileAttributes.AGE_OVER, String.class)).thenReturn(ageOverAttributes);
+
+        testObj.findAgeOverVerification(18);
+        AgeVerification result = testObj.findAgeOverVerification(18);
+
+        assertSame(ageOver18Attribute, result.getAttribute());
+        verify(profileMock, times(1)).findAttributesStartingWith(HumanProfileAttributes.AGE_UNDER, String.class);
```

Comment:
same here

Reply:
same response - it's particularly relevant, so I'd rather leave it